File: bindings/jvm/src/dev/qbt/Smoke.java
package dev.qbt;

public final class Smoke {
    private Smoke() {}

    public static void main(String[] args) throws Exception {
        String baseUrl = args.length > 0 ? args[0] : "http://127.0.0.1:8766";
        QbtClient client = new QbtClient(baseUrl);

        String health = client.health();
        if (!health.contains("\"status\": \"ok\"") && !health.contains("\"status\":\"ok\"")) {
            throw new IllegalStateException("QBT health contract failed: " + health);
        }

        String sample = client.sample("simulator", 128, 7);
        if (!sample.contains("\"active_sources\": 1")
                && !sample.contains("\"active_sources\":1")) {
            throw new IllegalStateException("QBT sample contract failed: " + sample);
        }

        String normalized = client.normalize(
                "{\"provider\":\"java\",\"backend\":\"smoke\",\"mode\":\"simulator\","
                        + "\"counts\":{\"0\":64,\"1\":64},\"shots\":128}");
        if (!normalized.contains("\"entropy\": 1.0")
                && !normalized.contains("\"entropy\":1.0")) {
            throw new IllegalStateException("QBT normalize contract failed: " + normalized);
        }

        System.out.println("Java/Kotlin QBT smoke: OK");
    }
}

File: bindings/perl/smoke.pl
use strict;
use warnings;
use HTTP::Tiny;
use JSON::PP qw(encode_json decode_json);

my $base = $ARGV[0] // 'http://127.0.0.1:8766';
$base =~ s{/$}{};
my $http = HTTP::Tiny->new(timeout => 5);

sub request_json {
    my ($method, $path, $payload) = @_;
    my %options = (headers => { 'Accept' => 'application/json' });
    if (defined $payload) {
        $options{headers}{'Content-Type'} = 'application/json';
        $options{content} = encode_json($payload);
    }
    my $response = $http->request($method, $base . $path, \%options);
    die "QBT HTTP $response->{status}: $response->{content}" unless $response->{success};
    return decode_json($response->{content});
}

my $health = request_json('GET', '/health', undef);
die 'QBT health contract failed' unless $health->{status} eq 'ok';

my $sample = request_json('POST', '/v1/sample', {
    provider => 'simulator', shots => 128, seed => 7,
});
die 'QBT sample contract failed' unless $sample->{packet}{active_sources} == 1;

my $normalized = request_json('POST', '/v1/normalize', {
    provider => 'perl', backend => 'smoke', mode => 'simulator',
    counts => { '0' => 64, '1' => 64 }, shots => 128,
});
die 'QBT normalize contract failed'
    unless abs($normalized->{state}{entropy} - 1.0) < 1e-12;

print "Perl QBT smoke: OK\n";
